### ISSUE TITLE
Performance drop when activating gradient clipping

### ISSUE THREAD

**Issue by Jonas-Jaeger** [PyTorchLightning/pytorch-lightning#1522]
Hello all,

I experienced a substantial drop in computation time when activating gradient clipping (by passing a non-zero value to the keyword argument `gradient_clip_val` when initializing the Trainer).

I noticed that in the current implementation of the `clipping_gradient` method in pytorch-lightning/trainer/training_tricks.py redundant computations are made by first computing the 2-norm and second squaring this result, which could be shortened by computing the sum of squares directly. This saves one square root and squaring operation per parameter set.

Best,
Jonas

### Environment

```
cuda:
	GPU:
	available:           False
	version:             None
packages:
	numpy:               1.18.1
	pyTorch_debug:       False
	pyTorch_version:     1.4.0
	pytorch-lightning:   0.7.4-dev
	tensorboard:         2.2.1
	tqdm:                4.45.0
system:
	OS:                  Darwin
	architecture:
		64bit
		
	processor:           i386
	python:              3.8.2
	version:             Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64
```

### Additional context

I trained a relatively small (two-layered) MLP on MNIST; perhaps this performance drop does not become that apparent when training on larger network architectures.

**Comment by Jonas-Jaeger** [comment]
Unfortunately, it seems that I do not have push access to this repository to push the proposed fix and create a pull request for this issue.

**Comment by williamFalcon** [comment]
fork the repo
make the fix
submit a PR
:)
